modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/rel/set/IgniteReduceIntersect.java

```diff
@@ -34,55 +34,68 @@
  * Physical node for REDUCE phase of INTERSECT operator.
  */
 public class IgniteReduceIntersect extends IgniteIntersect implements IgniteReduceSetOp {
+
+    private final int inputsNum;
```

Comment:
do we really need this as an explicit field? Is it possible compute this as difference `rowSize(inputOfReduce) - rowSize(reduce)`?

Reply:
Fixed.

Reply:
Reverted this change because this should be used in  `aggregateFieldsCount` (prior to that it was a constant):

>   public int aggregateFieldsCount() {
>     return rowType.getFieldCount() + inputsNum;
>   }

Reply:
This parameter is used by self cost estimation and we can not deduce it from the input relation, as there are no guarantees that there is no intermediate operators between `map` and `reduce` phases inserted by the optimiser.

Reply:
well, the truth is we don't care about intermediate operators, because they either are not changing row type -- then deduction will be correct, -- or plan is not correct at all, since reduce node is expecting rows to be in exact format defined by its MAP counterpart.

Seems like both `IgniteReduceIntersect` and `IgniteReduceMinus` should have `aggregateFieldsCount` implemented like 
```
    public int aggregateFieldsCount() {
        return getInput(0).getRowType().getFieldCount();
    }
```

Reply:
Fixed.